### DISCUSSION TITLE
Help! Want a toy example to run matmul with q40 weight by cuda kernel

### DISCUSSION THREAD

**Discussion by Eutenacity** [ggml-org/llama.cpp#9435] [Q&A] [answered]
Sorry, i am not familiar with the library, I want to run a matmul between a tensor created by pytorch and the q40 weight read from gguf.
I can read the weight from gguf and convert it to pytorch tensor.
But I have no idea to run the matmul between a tensor created by pytorch and the q40 weight by cuda kernel.

**Comment by slaren** [comment, marked as answer]
Some resources:
https://huggingface.co/blog/introduction-to-ggml
https://github.com/ggerganov/ggml/blob/master/examples/simple/simple-backend.cpp

**Reply by Eutenacity** [reply]
Is it possible to turn ggml_tensor to pytorch tensor on gpu without copy?
